ihmc-perception/src/main/java/us/ihmc/perception/gpuHeightMap/RapidHeightMapExtractorCuda.java

```diff
@@ -0,0 +1,543 @@
+      CUDATools.memcpyAsync(groundToWorldTransformDevicePointer, groundToWorldTransformHostPointer, groundToWorldTransformArray.length, stream);
+      CUDATools.memcpyAsync(parametersDevicePointer, parametersHostPointer, paramsArray.length, stream);
+
+      cudaStreamSynchronize(stream);
+
+
+      //Execute the CUDA kernels with the provided stream
+      //Each kernel performs a specific task related to the height map update, registration, and cropping
+
+      int blockSizeXY = 32;
+      int gridSizeKernel0X = (inputDepthImage.rows() + blockSizeXY - 1) / blockSizeXY;
+      int gridSizeKernel0Y = (inputDepthImage.cols() + blockSizeXY - 1) / blockSizeXY;
+      int gridSizeKernel1XY = (localCellsPerAxis + blockSizeXY - 1) / blockSizeXY;
+      int gridSizeKernel2XY = (globalCellsPerAxis + blockSizeXY - 1) / blockSizeXY;
+      int gridSizeKernel3XY = (heightMapParameters.getCropWindowSize() + blockSizeXY - 1) / blockSizeXY;
+
+      blockSize = new dim3(blockSizeXY, blockSizeXY, 1);
+      gridSizeKernel0 = new dim3(gridSizeKernel0X, gridSizeKernel0Y, 1);
+      gridSizeKernel1 = new dim3(gridSizeKernel1XY, gridSizeKernel1XY, 1);
+      gridSizeKernel2 = new dim3(gridSizeKernel2XY, gridSizeKernel2XY, 1);
+      gridSizeKernel3 = new dim3(gridSizeKernel3XY, gridSizeKernel3XY, 1);
+
+      preprocessKernel.withPointer(inputDepthImage.data()).withLong(inputDepthImage.step());
+      preprocessKernel.withPointer(transformedInputDepthImage.data()).withLong(transformedInputDepthImage.step());
+
+      updateKernel.withPointer(inputDepthImage.data()).withLong(inputDepthImage.step());
+      updateKernel.withPointer(localHeightMapImage.data()).withLong(localHeightMapImage.step());
+      updateKernel.withPointer(parametersDevicePointer);
+      updateKernel.withPointer(sensorToGroundTransformDevicePointer);
+      updateKernel.withPointer(groundToSensorTransformDevicePointer);
+
+      registerKernel.withPointer(localHeightMapImage.data()).withLong(localHeightMapImage.step());
+      registerKernel.withPointer(globalHeightMapImage.data()).withLong(globalHeightMapImage.step());
+      registerKernel.withPointer(parametersDevicePointer);
+      registerKernel.withPointer(worldToGroundTransformDevicePointer);
+      registerKernel.withPointer(sensorToGroundTransformDevicePointer);
+
+      croppingKernel.withPointer(globalHeightMapImage.data()).withLong(globalHeightMapImage.step());
+      croppingKernel.withPointer(sensorCroppedHeightMapImage.data()).withLong(sensorCroppedHeightMapImage.step());
+      croppingKernel.withPointer(parametersDevicePointer);
+      croppingKernel.withInt(heightMapParameters.getCropWindowSize());
+      cudaStreamSynchronize(stream);
+
+      preprocessKernel.run(stream, gridSizeKernel0, blockSize, 0);
+      updateKernel.run(stream, gridSizeKernel1, blockSize, 0);
+      cudaStreamSynchronize(stream);
```

Comment:
This shouldn't be needed (I'm not 100% sure on this, but I'd be very surprised if it was needed)